hadoop-pipeline-application/src/main/java/org/molgenis/hadoop/pipeline/application/sequences/AlignedReadType.java

```diff
@@ -0,0 +1,103 @@
+package org.molgenis.hadoop.pipeline.application.sequences;
+
+import java.util.List;
+
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import htsjdk.samtools.SAMRecord;
+
+/**
+ * The type of an {@link AlignedRead}.
+ */
+public enum AlignedReadType
+{
+	/**
+	 * The read did not map.
+	 */
+	UNMAPPED,
+
+	/**
+	 * The read mapped once.
+	 */
+	MAPPED,
+
+	/**
+	 * The read mapped and secondary/supplementary read(s) as well.
+	 */
+	MULTIMAPPED,
+
+	/**
+	 * Only a secondary/supplementary read(s) aligned, but not the primary read.
+	 */
+	MULTIMAPPED_SUPPLEMENTARY_ONLY,
+
+	/**
+	 * The data is deemed invalid.
+	 */
+	INVALID;
+
+	/**
+	 * Determines the {@link AlignedReadType} of a {@link List}{@code <}{@link SAMRecord}{@code >}. If the
+	 * {@link SAMRecord} {@link List} does not adhere to the requirements, returns {@link #INVALID} as type. Otherwise,
+	 * returns one of the other possible types.
+	 * 
+	 * @param records
+	 *            {@link List}{@code <}{@link SAMRecord}{@code >}
+	 * @return {@link AlignedReadType}
+	 */
+	static AlignedReadType determineType(List<SAMRecord> records)
+	{
+		// When list is empty, type is INVALID.
+		if (records.size() == 0) return INVALID;
+
+		// Retrieves primary record and also checks whether there is more than 1 primary record.
+		SAMRecord primaryRecord = null;
+		for (SAMRecord record : records)
+		{
+			if (!record.isSecondaryOrSupplementary())
+			{
+				// If there is more than 1 primary record, returns INVALID.
+				if (primaryRecord != null) return INVALID;
```

Comment:
`AlignedRead.getRecords()` should in theory only contain one primary record. However, scenarios such as using a binary tool for alignment that generates unexpected output or does not correctly tag secondary/supplementary SAMRecords might result in more records that are tagged as primary. Due to the fact that `AlignedReadType.determineType()` is able to be called outside of `AlignedRead`, no assumption can be made that the `List<SAMRecord>` given to `AlignedReadType.determineType(List<SAMRecord>)` is sorted by default (in which case only the first two records would need to be looked at).

While in this version it might not be needed, if for example someone would use a custom tools archive (with different versions of the used tools and therefore a chance for unexpected behavior) or the design would use a different tool instead (which might not tag all records in the same way), this implementation provides a safeguard for unexpected data retrieved from the binary tool output.

Therefore, I will adjust the design so that the Type enums for both `AlignedRead` and `AlignedReadPair` will be converted into inner enums so that the `determineType()` methods in both can be made `private`.